commands/translate.py

```diff
@@ -0,0 +1,169 @@
+            flags = {}
+
+        flags[emoji] = langcode
+
+        # Write the formats to the JSON file
+        with open(FLAGS_FILE, 'w') as new:
+            json.dump(flags, new)
+
+        return "%s now maps to %s!" % (emoji, code(langcode))
+
+
+class Remove(Translate):
+    desc = "Removes a emoji->langcode mapping if it exists. Mods only."
+    roles_required = ['mod', 'exec']
+
+    def eval(self, emoji):
+        # Open the JSON file or create a new dict to load
+        try:
+            with open(FLAGS_FILE, 'r') as old:
+                flags = json.load(old)
+            langcode = flags.pop(emoji)
+
+        except (FileNotFoundError, KeyError, ValueError):
+            raise CommandFailure("Emoji %s not found!" % emoji)
+
+        # Write the formats to the JSON file
+        with open(FLAGS_FILE, 'w') as new:
+            json.dump(flags, new)
+
+        return "Mapping of %s to %s removed!" % (emoji, code(langcode))
+
+class List(Translate):
+    desc = "Lists all a emoji->langcode mappings. Mods only."
+    roles_required = ['mod', 'exec']
+
+    def eval(self):
+        # Open the JSON file, skip if it does not exist
+        try:
+            with open(FLAGS_FILE, 'r') as old:
+                flags = json.load(old)
+        except FileNotFoundError:
+            raise CommandFailure("Emoji list is empty!")
+
+        return EmbedTable(fields=['Emoji->Code'], 
+                         table=[(emoji+'->'+code,) for emoji, code in flags.items()], 
+                         colour=TRANSLATE_COLOR)
+
+
+class Entry():
+    def __init__(self, message, language, flag):
+        self.author = message.author
+        self.content = message.clean_content
+        self.timestamp = message.timestamp
+        self.language = language
+        self.flag = flag
+
+
+    def as_embed(self, title):
+        embed = Embed(description=self.content,
+                     colour=self.author.colour,
+                     timestamp=self.timestamp)
+        embed.set_author(name=nick(self.author),
+                         icon_url=AVATAR_FORMAT.format(self.author))
+        embed.set_footer(text=title)
+        return embed
+
+
+async def create_list(logs):
+    i = 0
+    archive = []
+
+    # load dict of flags
+    try:
+        with open(FLAGS_FILE, 'r') as f:
+            flags = json.load(f)
+    except FileNotFoundError:
+        raise CommandFailure('Flags file does not exist!')
+
+    # Iterate through message history
+    async for message in logs:
+        # Break when we've found HISTORY messages
+        if i >= HISTORY:
+            break
+
+        reactions = [x.emoji for x in message.reactions]
+        lang_list = []
+        for r in reactions:
+            if str(r) in flags:
+                lang_list.append(Entry(message, flags[str(r)], r))
+                i += 1
+
+        if lang_list:
+            archive.append(lang_list)
+                
+    return archive
+
+
+def removeEmojis(text):
```

Comment:
Doesn't work with stacked emojis:

BEFORE:
this :wave: :wave: :cowboy:  ain't :cow: my :boy:  first rodeo :rose: :cowboy: :bulb:

AFTER:
this   🤠  ain't  my   first rodeo  🤠